fabtests/common/shared.c

```diff
@@ -1701,7 +1701,7 @@ int ft_read_addr_opts(char **node, char **service, struct fi_info *hints,
 {
 	int ret;
 
-	if (opts->dst_addr && (opts->src_addr || !opts->oob_port)){
+	if (opts->dst_addr) {
```

Comment:
Not exactly sure why, but this change is the reason AWS CI EFA test failed.

Reply:
The additional checks can result in the code now taking the else path even when we have a dst_addr.  That sets FI_SOURCE and changes the behavior of fi_getinfo.  Something is off about the logic here.